vm/proxyapp/proxyappclient.go

```diff
@@ -151,19 +206,16 @@ func runProxyApp(params *proxyAppParams, cmd string) (*ProxyApp, error) {
 		cancelContext()
 	}
 
-	subStdout, err := subProcess.StdoutPipe()
-	if err != nil {
-		freeAll()
-		return nil, fmt.Errorf("failed to get stdoutpipe: %w", err)
-	}
-	toClose = append(toClose, subStdout)
-
-	subStdin, err := subProcess.StdinPipe()
-	if err != nil {
-		freeAll()
-		return nil, fmt.Errorf("failed to get stdinpipe: %w", err)
+	var client *rpc.Client
+	if initRPClient {
+		var err error
+		client, err = initPipedRPCClient(subProcess)
+		if err != nil {
+			freeAll()
+			return nil, fmt.Errorf("failed to init piped client: %w", err)
+		}
+		toClose = append(toClose, client)
```

Comment:
Hummm... now we will close the pipes closeProxy method as part of p.proxy.Client.Close(), right? While previously we have not. What behavior is correct, the old one of the new one?

Docs say:
```
Wait will close the pipe after seeing the command exit
````
https://pkg.go.dev/os/exec#Cmd.StdoutPipe

If I am reading this correctly, here:

```
func (p *pool) closeProxy() {
...
		if p.proxy.Client != nil {
			p.proxy.Client.Close()
		}
		if p.proxy.terminate != nil {
			p.proxy.terminate()
			<-p.proxy.onTerminated
		}
...
}
```

we now have 2 racing closes of the pipes: one happens as part of Client.Close() and the second concurrently as part of cmd.Wait().

Reply:
Fixed. PTAL.